File: javascript/iterate_array.js
const buildFibonacciSequence = (maxItems) => {
  const array = Array.from(Array(maxItems).keys());

  let actual = 0;
  let previous = null;
  return array.map((_) => {
    if (previous === null) {
      previous = 1;
      actual = 0
      return 0;
    }

    const sum = actual + previous;
    previous = actual;
    actual = sum;
    return sum;
  });
};

module.exports = buildFibonacciSequence;

File: python3/iterate_array.py
def buildFibonacciSequence(items):
    actual = 0
    previous = None
    sequence = []
    for index in range(items):
        if(previous == None):
            previous = 1
            actual = 0
            sequence.append(0)
            continue
        
        
        sum = actual + previous 
        previous = actual
        actual = sum
        sequence.append(sum)

    return sequence
